### PR TITLE
fix: Unordered tables in postgresql_publication

### PR DESCRIPTION
**Pull request by nguyenhoaibao**
Fixes https://github.com/cyrilgdn/terraform-provider-postgresql/issues/218

### CODE REVIEW

**Review by cyrilgdn — CHANGES_REQUESTED** [review]
Hi @nguyenhoaibao ,

Thanks for your PR.

This fix works only if you sorted the table in the TF definition.
I think that the right fix would be to switch `tables` from `TypeList` to `TypeSet` which is actually an unordered array.

Would you want to give it a try?

I think we just need to change the type, then cast with `tables.(*schema.Set)` instead of `tables.([]interface{})` and when needed (to `range` over the list for example), we can call the `tables.(*schema.Set).List()` function of Set to get an array.

Also it would be great to tweak the tests so they fail without the fix.

**Comment by nguyenhoaibao** [comment]
@cyrilgdn thanks for your reviewing. Let me try with your suggestion.

**Comment by nguyenhoaibao** [comment]
@cyrilgdn I made a change using your suggestion. Please take a look.

File: postgresql/resource_postgresql_publication.go [hunk 1]
```diff
@@ -372,13 +372,10 @@ func resourcePostgreSQLPublicationReadImpl(db *DBConnection, d *schema.ResourceD
 		}
 		tables = append(tables, table)
 	}
-
-	switch {
```
Comment (nguyenhoaibao) [inline postgresql/resource_postgresql_publication.go:376]: I see these error handling are redundant so I removed as well.

File: postgresql/resource_postgresql_publication.go [hunk 2]
```diff
@@ -372,13 +372,10 @@ func resourcePostgreSQLPublicationReadImpl(db *DBConnection, d *schema.ResourceD
...
-	case err == sql.ErrNoRows:
-		log.Printf("[WARN] No PostgreSQL tables found for Publication %s", PublicationName)
-	case err != nil:
-		return fmt.Errorf("Error reading Publication tables: %w", err)
```
Comment (nguyenhoaibao) [inline postgresql/resource_postgresql_publication.go:380]: I see these error handling are redundant so I removed as well.

File: postgresql/resource_postgresql_publication.go [hunk 3]
```diff
@@ -372,13 +372,10 @@ func resourcePostgreSQLPublicationReadImpl(db *DBConnection, d *schema.ResourceD
...
+	if err := rows.Err(); err != nil {
```
Comment (nguyenhoaibao) [inline postgresql/resource_postgresql_publication.go:375]: I added this error handling here to catch any error that can happen during the rows.Next iteration, as described in the  doc https://pkg.go.dev/database/sql#Rows.Err

**Review by cyrilgdn — APPROVED** [review]
Thanks :+1:

**Pull request merged**